File: scripts/mcp_launcher.py
import os
import sys
from pathlib import Path


ROOT = Path(__file__).resolve().parents[1]


def virtualenv_python(root=ROOT):
    candidates = [
        Path(root) / ".venv" / "Scripts" / "python.exe",
        Path(root) / ".venv" / "bin" / "python",
    ]
    return next((path for path in candidates if path.is_file()), None)


def main():
    python = virtualenv_python()
    if python is None:
        raise SystemExit(
            "Job Search Copilot is not set up. Run "
            "`python scripts/bootstrap.py` from the repository root."
        )
    os.chdir(ROOT)
    os.execv(str(python), [str(python), str(ROOT / "mcp_server.py")])


if __name__ == "__main__":
    main()

File: scripts/mcp_launcher.js
const { spawnSync } = require("node:child_process");
const fs = require("node:fs");
const path = require("node:path");

const ROOT = path.resolve(__dirname, "..");

function virtualenvPython(root = ROOT) {
  const candidates = [
    path.join(root, ".venv", "Scripts", "python.exe"),
    path.join(root, ".venv", "bin", "python"),
  ];
  return candidates.find((candidate) => fs.existsSync(candidate));
}

function main(root = ROOT) {
  const python = virtualenvPython(root);
  if (!python) {
    console.error(
      "Job Search Copilot is not set up. Run `python scripts/bootstrap.py` " +
        "from the repository root.",
    );
    return 1;
  }

  const result = spawnSync(python, [path.join(root, "mcp_server.py")], {
    cwd: root,
    stdio: "inherit",
  });
  if (result.error) {
    console.error(`Unable to start Job Search Copilot MCP: ${result.error.message}`);
    return 1;
  }
  return result.status ?? 1;
}

if (require.main === module) {
  process.exitCode = main();
}

module.exports = { main, virtualenvPython };
